daffodil-runtime2/src/main/resources/org/apache/daffodil/runtime2/c/libcli/daffodil_main.c

```diff
@@ -15,13 +15,13 @@
  * limitations under the License.
  */
 
-#include <stdio.h>          // for NULL, perror, FILE, fclose, fflush, fopen, stdin, stdout
-#include <string.h>         // for strcmp
-#include "daffodil_argp.h"  // for daffodil_parse, daffodil_parse_cli, daffodil_unparse, daffodil_unparse_cli, daffodil_cli, parse_daffodil_cli, DAFFODIL_PARSE, DAFFODIL_UNPARSE
-#include "errors.h"         // for continue_or_exit, Error, print_diagnostics, PState, UState, ERR_FILE_CLOSE, ERR_FILE_FLUSH, ERR_FILE_OPEN, ERR_INFOSET_READ, ERR_INFOSET_WRITE
-#include "infoset.h"        // for walkInfoset, InfosetBase, rootElement, ERD, VisitEventHandler
-#include "xml_reader.h"     // for xmlReaderMethods, XMLReader
-#include "xml_writer.h"     // for xmlWriterMethods, XMLWriter
+#include <stdio.h>            // for NULL, FILE, perror, fclose, fopen, stdin, stdout
+#include <string.h>           // for strcmp
+#include "daffodil_getopt.h"  // for daffodil_cli, parse_daffodil_cli, daffodil_parse, daffodil_parse_cli, daffodil_unparse, daffodil_unparse_cli, DAFFODIL_PARSE, DAFFODIL_UNPARSE
```

Comment:
Line too long

Reply:
What about just removing those comments? If the includes are autogenerated, is the information about why they are added useful? Seems it's only useful if you don't trust the autogenerated includes and want to verify why it added lines, which doesn't seem that necessary.

Reply:
No, I don't want to remove the comments.  The whole point of the comments is for me to show a developer looking at the code exactly what is being used from each include header.  If a developer is curious to learn more about a symbol, he or she can find exactly where to look by searching the comments to find the right include header and then open that .h file (also the .c file if both are in the repository).

The includes are not autogenerated.  The file has to compile successfully before iwyu can process it and iwyu doesn't edit the file.  It only prints some replacement includes / comments on standard output along with some other output and I have to manually edit the file to replace the includes / comments.  In fact, iwyu won't even print the replacement includes / comments if the only changes would be to the comments (it has no way of knowing what the original comments were), so I have to manually add an extra include in order for iwyu to show me the updated comments.  I wouldn't bother using iwyu unless it was to put these comments there as well as get rid of unneeded includes.

Reply:
I don't have any problem with these comments, that said,

> If a developer is curious to learn more about a symbol, he or she can find exactly where to look by searching the comments to find the right include header and then open that .h file (also the .c file if both are in the repository).

Most IDE's and even text editors used for coding have an easy way to jump directly to the definition of a symbol. ctags+vim is my go-to, but there's lots of options for a more efficient alternative than searching through #include comments.

Your workflow to update these comments also seems pretty laborious. If it's not automated, I would expect that once other developers get involved that aren't used to this workflow it's going to be hard to keep these comments updated and useful. The removal of superfluous #includes does seem like a useful feature of iwyu though (I wasn't aware of this tool). Seems like there might be some benefit it automating some of these capabilities.

FYI, I've spoken to some devs recently that sound interested in this runtime2 and have similar FPGA goals, hoping to get them involved soon.